### PR TITLE
Catch thread in example

### PR DESCRIPTION
**Pull request by dns2utf8**
Since this is an example, the code will be copied by many people and should be over correct.

?r @steveklabnik

### CODE REVIEW

**Comment by brson** [comment]
The example looks correct as written to me. Why join it?

**Comment by durka** [comment]
The example as amended in this PR will throw a "must use Result" warning.

File: src/libcore/sync/atomic.rs [hunk 1]
```diff
@@ -48,12 +48,14 @@
...
 //!     let spinlock_clone = spinlock.clone();
-//!     thread::spawn(move|| {
+//!     let thread = thread::spawn(move|| {
 //!         spinlock_clone.store(0, Ordering::SeqCst);
 //!     });
 //!
 //!     // Wait for the other thread to release the lock
 //!     while spinlock.load(Ordering::SeqCst) != 0 {}
+//!
+//!     thread.join();
```
Comment (durka) [inline src/libcore/sync/atomic.rs]: This line will get a "must use Result" warning -- if the idea is to put good code in examples, it should be handled.

**Comment by steveklabnik** [comment]
I agree with @brson here: unless you're specifically doing something with the `thread` variable, why explicitly join it?

**Comment by dns2utf8** [comment]
I had the situation, where the inner thread outlived the outer and caused unexpeced behavior: [Run on play](https://play.rust-lang.org/?code=use%20std%3A%3Async%3A%3AArc%3B%0Ause%20std%3A%3Async%3A%3Aatomic%3A%3A%7BAtomicUsize%2C%20Ordering%7D%3B%0Ause%20std%3A%3Athread%3B%0Ause%20std%3A%3Atime%3A%3ADuration%3B%0A%0Afn%20main()%20%7B%0A%20%20%20%20let%20spinlock%20%3D%20Arc%3A%3Anew(AtomicUsize%3A%3Anew(1))%3B%0A%0A%20%20%20%20let%20spinlock_clone%20%3D%20spinlock.clone()%3B%0A%20%20%20%20let%20thread%20%3D%20thread%3A%3Aspawn(move%7C%7C%20%7B%0A%20%20%20%20%20%20%20%20spinlock_clone.store(0%2C%20Ordering%3A%3ASeqCst)%3B%0A%20%20%20%20%20%20%20%20%0A%20%20%20%20%20%20%20%20thread%3A%3Asleep(Duration%3A%3Afrom_secs(2))%3B%0A%20%20%20%20%20%20%20%20for%20i%20in%200..10%20%7B%0A%20%20%20%20%20%20%20%20%20%20%20%20println!(%22%7B%7D%22%2C%20i)%3B%0A%20%20%20%20%20%20%20%20%7D%0A%20%20%20%20%7D)%3B%0A%0A%20%20%20%20%2F%2F%20Wait%20for%20the%20other%20thread%20to%20release%20the%20lock%0A%20%20%20%20while%20spinlock.load(Ordering%3A%3ASeqCst)%20!%3D%200%20%7B%7D%0A%0A%20%20%20%20%2F%2F%20thread.join().is_ok()%3B%0A%20%20%20%20println!(%22END%22)%3B%0A%7D&version=stable&backtrace=0)
```rust
use std::sync::Arc;
use std::sync::atomic::{AtomicUsize, Ordering};
use std::thread;
use std::time::Duration;

fn main() {
    let spinlock = Arc::new(AtomicUsize::new(1));

    let spinlock_clone = spinlock.clone();
    let thread = thread::spawn(move|| {
        spinlock_clone.store(0, Ordering::SeqCst);
        
        thread::sleep(Duration::from_secs(2));
        for i in 0..10 {
            println!("{}", i);
        }
    });

    // Wait for the other thread to release the lock
    while spinlock.load(Ordering::SeqCst) != 0 {}

    // thread.join().is_ok();
    println!("END");
}
```

I could also add a hint to the threading model of rust, but I could not find a reference explaining what happens in this case with the inner thread.

**Comment by durka** [comment]
Yeah, the main thread exits as soon as the inner thread releases the lock (and the program ends when the main thread exits) so the numbers are never printed. If the `spinlock_clone.store` line is moved until after the loop, then the numbers are indeed printed.

As for using the Result, calling `is_ok` and not looking at the return value will technically silence the warning but is actually no better.

**Comment by steveklabnik** [comment]
@dns2utf8 ahhh yes, I can see that. Hmmm. (and @durka's explanation is correct)

**Comment by dns2utf8** [comment]
@durka you are right about `is_ok`.

I am thinking about a link to `std::thread#The treading model` in the text section.

**Comment by dns2utf8** [comment]
Hm, I seem to have broken the build with the link. Do you have an idea?

File: src/libcore/sync/atomic.rs [hunk 2]
```diff
@@ -26,7 +26,9 @@
 //! [1]: http://llvm.org/docs/LangRef.html#memory-model-for-concurrent-operations
 //!
 //! Atomic variables are safe to share between threads (they implement `Sync`)
-//! but they do not themselves provide the mechanism for sharing. The most
+//! but they do not themselves provide the mechanism for sharing and follow the
```
Comment (steveklabnik) [inline src/libcore/sync/atomic.rs]: this line has a trailing whitespace, that's what's failing the build

Comment (dns2utf8) [inline src/libcore/sync/atomic.rs]: Ahh, thanks :+1:

**Comment by dns2utf8** [comment]
Thanks @steveklabnik, I rebased it to current master.

**Pull request merged**